src/util/simplify_expr_pointer.cpp

```diff
@@ -388,41 +388,101 @@ bool simplify_exprt::simplify_pointer_offset(exprt &expr)
   return true;
 }
 
+static bool simple_structural_lvalue_equal(const exprt &lhs, const exprt &rhs)
```

Comment:
Isn't this equivalent to just using the `==` operator after checking that something is either a `symbol` and or a `member`

Reply:
what about situations like `&(*random_S()).x == &(*random_S()).x`?

Reply:
Good point!